pkg/urbit/noun/serial.c

```diff
@@ -70,12 +71,51 @@ _cs_jam_fib_mat(struct _cs_jam_fib* fib_u, u3_noun a)
     _cs_jam_fib_chop(fib_u, 1, 1);
   }
   else {
-    c3_w a_w = u3r_met(0, a);
-    c3_w b_w = c3_bits_word(a_w);
+    c3_w   a_w = u3r_met(0, a);
+    c3_w   b_w = c3_bits_word(a_w);
+    c3_w bit_w = fib_u->bit_w;
+
+    //  amortize overflow checks and reallocation
+    //
+    {
+      c3_w met_w = a_w + (2 * b_w);
+
+      if ( a_w > (UINT32_MAX - 64) ) {
+        u3m_bail(c3__fail);
+        return;
+      }
```

Comment:
Can you add a comment explaining that `a_w` must be less than or equal to `UINT32_MAX - 64` so that there's space for `2 * b_w`, which is at most 64?

Reply:
The line immediately prior to this calculates `a_w + (2 * b_w)`, `b_w` is the result of `c3_bits_word()`, and a word is 32 bits. There's no end to the amount of documentation required if all such relationships need be spelled out in prose.

Reply:
It takes one line, literally 7 tokens. Put it inline for all I care. Again, the relationship between some of these things is obvious in the head of whoever is writing the code, but it's not immediately obvious to someone reading the code.
```c
// 2 * b_w is at most 64
```